oak_dice/src/evidence.rs

```diff
@@ -74,8 +82,29 @@ pub struct RootLayerEvidence {
 }
 
 impl RootLayerEvidence {
-    pub fn get_tee_platform(&self) -> Option<TeePlatform> {
-        TeePlatform::from_repr(self.tee_platform)
+    pub fn get_tee_platform(&self) -> Result<TeePlatform, &'static str> {
+        TeePlatform::from_repr(self.tee_platform).ok_or("invalid TEE Platform value")
+    }
+
+    pub fn get_remote_attestation_report(&self) -> Result<&[u8], &'static str> {
+        match self.get_tee_platform()? {
+            TeePlatform::AmdSevSnp => {
+                Ok(&self.remote_attestation_report[..AMD_SEV_SNP_ATTESTATION_REPORT_SIZE])
+            }
+            _ => Ok(&self.remote_attestation_report),
```

Comment:
Do we really want to return anything from that field in this case? That seems risky. Should we panic / return an error instead?